package/android/src/main/java/com/mrousavy/camera/CameraView+Events.kt

```diff
@@ -87,14 +94,23 @@ fun CameraView.invokeOnCodeScanned(barcodes: List<Barcode>, scannerFrame: CodeSc
     codes.pushMap(code)
   }
 
-  val event = Arguments.createMap()
-  event.putArray("codes", codes)
+  val data = Arguments.createMap()
+  data.putArray("codes", codes)
   val codeScannerFrame = Arguments.createMap()
   codeScannerFrame.putInt("width", scannerFrame.width)
   codeScannerFrame.putInt("height", scannerFrame.height)
-  event.putMap("frame", codeScannerFrame)
+  data.putMap("frame", codeScannerFrame)
+
+  val surfaceId = UIManagerHelper.getSurfaceId(this)
+  val event = CameraCodeScannedEvent(surfaceId, id, data)
+  this.sendEvent(event)
+}
+
+private fun CameraView.sendEvent(event: Event<*>) {
   val reactContext = context as ReactContext
-  reactContext.getJSModule(RCTEventEmitter::class.java).receiveEvent(id, "cameraCodeScanned", event)
+  val dispatcher =
+    UIManagerHelper.getEventDispatcherForReactTag(reactContext, id)
+  dispatcher?.dispatchEvent(event)
```

Comment:
Do we wanna throw if this cannot be found? Or just ignore it? 🤔

Reply:
I think it's safe to ignore this:
- this should not happen, something must go very wrong so that the event dispatcher is not available
- in this case, React Native itself will [show an exception](https://github.com/facebook/react-native/blob/a80674fc7c04da5c8630dc4bfc63904fd38e9e71/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/uimanager/UIManagerHelper.java#L107)